xapian-letor/feature/colltfcolllenfeature.cc

```diff
@@ -0,0 +1,71 @@
+/** @file colltfcolllenfeature.cc
+ * @brief CollTfCollLenFeature class
+ */
+/* Copyright (C) 2012 Parth Gupta
+ * Copyright (C) 2016 Ayush Tomar
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+
+#include "xapian-letor/feature.h"
+#include "feature_internal.h"
+
+using namespace std;
+
+namespace Xapian {
+
+std::string
+CollTfCollLenFeature::name() {
+    return "CollTfCollLenFeature";
+}
+
+double
+CollTfCollLenFeature::get_fval(char ch) {
+
+    Query query = Feature::internal->feature_query;
+    map<string, long int> coll_tf = Feature::internal->collection_termfreq();
+    map<string, long int> coll_len = Feature::internal->collection_length();
+
+    double value = 0;
+
+    if (ch == 't') {
+    for (Xapian::TermIterator qt = query.get_terms_begin();
```

Comment:
Indentation is weird throughout this bit, making it hard to read (at least on github).